### PR TITLE
Handling continuation token and Max results param from listobjects request and setting continuation token in response

### PR DESCRIPTION
**Pull request by Tulsishah**

### CODE REVIEW

File: internal/storage/bucket_handle.go [hunk 1]
```diff
@@ -210,11 +210,16 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
...
 	pi := itr.PageInfo()
 	pi.MaxSize = req.MaxResults
+	pi.Token = req.ContinuationToken
 	var list gcs.Listing
 
 	// Iterating through all the objects in the bucket and one by one adding them to the list.
 	for {
 		var attrs *storage.ObjectAttrs
+		if len(list.Objects) == req.MaxResults {
+			err = nil
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:220]: 1. add a comment why this if condition is required
2. err is nil by default. Do we still need to set err=nil?

File: internal/storage/bucket_handle_test.go [hunk 2]
```diff
@@ -353,3 +353,21 @@ func (t *BucketHandleTest) TestListObjectMethodWithEmptyDelimiter() {
 	AssertEq(TestObjectGeneration, obj.Objects[0].Generation)
 	AssertEq(nil, obj.CollapsedRuns)
 }
+
+func (t *BucketHandleTest) TestListObjectMethodForMaxResult() {
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle_test.go:357]: how many objects did you add in the fakeserver?

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]

File: internal/storage/bucket_handle_test.go (hunk 2 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle_test.go:357]: four

File: internal/storage/bucket_handle.go (hunk 1 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: fixed

File: internal/storage/bucket_handle_test.go (hunk 2 above)
Comment (vadlakondaswetha) [reply internal/storage/bucket_handle_test.go:357]: add a comment here that we have 4 objects in fakeserver

File: internal/storage/bucket_handle.go [hunk 3]
```diff
@@ -210,14 +210,18 @@ func (b *bucketHandle) ListObjects(ctx context.Context, req *gcs.ListObjectsRequ
 	itr := b.bucket.Objects(ctx, query) // Returning iterator to the list of objects.
 	pi := itr.PageInfo()
 	pi.MaxSize = req.MaxResults
+	pi.Token = req.ContinuationToken
 	var list gcs.Listing
 
 	// Iterating through all the objects in the bucket and one by one adding them to the list.
 	for {
 		var attrs *storage.ObjectAttrs
+		// Fetching number of objects equals to requested MaxResults
```
Comment (vadlakondaswetha) [inline internal/storage/bucket_handle.go:219]: Please update the comment like this. itr.next returns all the objects present in the bucket. Hence adding a check to break after required number of objects are returned.

**Review by vadlakondaswetha — CHANGES_REQUESTED** [review]
Please update the pr description.

File: internal/storage/bucket_handle_test.go (hunk 2 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle_test.go:358]: fixed

File: internal/storage/bucket_handle.go (hunk 3 above)
Comment (Tulsishah) [reply internal/storage/bucket_handle.go]: fixed

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**